src/ksync/ksync_sock_user.cc

```diff
@@ -109,15 +108,18 @@ void KSyncSockTypeMap::PurgeTxBuffer() {
     }
 
     // If there are more than one NETLINK messages, we need to add 
-    struct nlmsghdr nlh;
     if (count > 1) {
         //Send Netlink-Done message NLMSG_DONE at end
+        struct nlmsghdr nlh;
         InitNetlinkDoneMsg(&nlh, last_nlh->nlmsg_seq);
         iovec.push_back(buffer((uint8_t *)&nlh, NLMSG_HDRLEN));
     } else {
         // Single buffer. Reset the MULTI flag
-        if (last_nlh)
-            last_nlh->nlmsg_flags &= (~NLM_F_MULTI);
+        assert(iovec.size() == 1);
+
+        char *buf = boost::asio::buffer_cast<char *>(iovec.back());
+        struct nlmsghdr *nlh = (struct nlmsghdr *)buf;
+        nlh->nlmsg_flags &= (~NLM_F_MULTI);
```

Comment:
Because `last_nlh` is still used in the loop before this condition, we could use it instead of manually getting last header. So old code can be used.